src/BeamAdapter/component/mapping/AdaptiveBeamMapping.inl

```diff
@@ -45,6 +45,20 @@
 #include <sofa/core/ConstraintParams.h>
 #include <sofa/core/MechanicalParams.h>
 
+// execution policies only supported by MSVC >=2019 and GCC >=10
+#ifdef _MSC_VER
+    #define HAS_SUPPORT_STL_PARALLELISM (_MSC_VER > 1921)
+#elif defined(__GNUC__)
+    #define HAS_SUPPORT_STL_PARALLELISM  (__GNUC__ > 9)
+#else
+    #define HAS_SUPPORT_STL_PARALLELISM  false
+#endif
+
+
+#if HAS_SUPPORT_STL_PARALLELISM
+#include <execution>
+#endif
```

Comment:
Nope, this is a feature provided by the STL/C++17, which allows us an easy-to-use way to get parallel/unordered computations, mostly for parallelize for-loop (at least for my level)
As far as I know, Multithreading and some other stuff in SOFA use explicit thread usage/pool etc. at an higher level. (using boost or std::thread, etc) so more for parallelizing complex algorithms, etc

Reply:
I wonder if `HAS_SUPPORT_STL_PARALLELISM` is not simply = `__has_include(<execution>)`. What do you think? It would avoid to rely on the compiler version. It would also be compatible with clang when (if) parallelism will be supported.

Reply:
Unfortunately no, gcc < 10 (9 and some other versions ) supports `<execution>` but you need to find_package(tbb) and link with it. Otherwise gcc will complain that it cannot include `tbb/something.h` 
And I did not want to a special case for a "old" gcc in cmake 